src/components/Proposal/ProposalDetailsPage.tsx

```diff
@@ -97,6 +97,28 @@ class ProposalDetailsPage extends React.Component<IProps, IState> {
     this.setState({ showVotersModal: false });
   }
 
+  private parseYouTubeVideoIdFromUri = (url: string): string => {
+    const videoid = url.match(/(\/|%3D|v=)([0-9A-z-_]{11})([%#?&]|$)/);
+    if(videoid && (videoid.length >= 3)) {
+      return videoid[2];
+    } else { 
+      // eslint-disable-next-line no-console
+      console.error("The youtube url is not valid.");
+      return null;
+    }
+  }
+
+  private getVimeoIdFromUrl = (url: string): string => {
+    const videoid = url.match(/^.*(?:vimeo.com)\/(?:channels\/|channels\/\w+\/|groups\/[^/]*\/videos\/|album\/\d+\/video\/|video\/|)(\d+)(?:$|\/|\?)/);
+    if(videoid && (videoid.length >= 2)) {
+      return videoid[1];
+    } else { 
+      // eslint-disable-next-line no-console
+      console.error("The youtube url is not valid.");
```

Comment:
Vimeo

Reply:
fixed, and note I also improved the error handling logic